### PR TITLE
[GSoC2024] add return types to fix typescript warnings

### PR DESCRIPTION
**Pull request by tahamukhtar20**
Fixes: #7598
### Motivation and context
There are some minor typescript errors in the files, this pull request fixes those.

### How has this been tested?
Ran the code after changing, and it was functional like before, this didn't need any additional test.

### Checklist

- [x] I submit my changes into the `develop` branch
- [x] I have created a changelog fragment 
- [x] I have linked related issues (see [GitHub docs](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))
  ([cvat-canvas](https://github.com/opencv/cvat/tree/develop/cvat-canvas#versioning),
  [cvat-core](https://github.com/opencv/cvat/tree/develop/cvat-core#versioning),
  [cvat-data](https://github.com/opencv/cvat/tree/develop/cvat-data#versioning) and
  [cvat-ui](https://github.com/opencv/cvat/tree/develop/cvat-ui#versioning))

### License

- [x] I submit _my code changes_ under the same [MIT License](
  https://github.com/opencv/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.

### CODE REVIEW

**Comment by tahamukhtar20** [comment]
@bsekachev I've made the necessary changes, please change the last file Lambda Manager as I have added a few Promise<any> there because it is a very reused file and to me, Promise<any> makes the most sense there. Please check and let me know if there is any issue and I'll fix it ASAP. Thanks.

**Comment by tahamukhtar20** [comment]
I think this PR is not ready yet, there are a few changes which I just realized need to be added, I'll add those and ping you accordingly, thanks.

File: cvat-core/src/lambda-manager.ts [hunk 1]
```diff
@@ -43,7 +43,7 @@ class LambdaManager {
         return { models, count: lambdaFunctions.length };
     }
 
-    async run(taskID: number, model: MLModel, args: any) {
```
Comment (bsekachev) [inline cvat-core/src/lambda-manager.ts:46]: Promise<any> should be specified more concrete in this file

File: changelog.d/20240327_224004_tahamukhtar20.md [hunk 2]
```diff
@@ -0,0 +1,4 @@
+### Fixed
+
+- Fixed typescript linting errors in ts files
```
Comment (SpecLad) [inline changelog.d/20240327_224004_tahamukhtar20.md:3]: This fix has no impact on the user, so I don't think it needs a changelog entry.

File: cvat-core/src/lambda-manager.ts (hunk 1 above)
Comment (bsekachev) [reply cvat-core/src/lambda-manager.ts:46]: @tahamukhtar20

Comment (tahamukhtar20) [reply cvat-core/src/lambda-manager.ts:46]: I'll get to these tomorrow after submitting my proposal. Thanks for the reminder.

**Comment by bsekachev** [comment]
Hello, do you have any plans regarding the pull request?

**Comment by tahamukhtar20** [comment]
Yes @bsekachev , I left a comment above under your review, please let me know. Thanks.

**Comment by bsekachev** [comment]
Yes, one week ago you mentioned "tomorrow". That is why I am asking :)

**Comment by tahamukhtar20** [comment]
> Yes, one week ago you mentioned "tomorrow". That is why I am asking :)

Sorry about that, @bsekachev  I wanted to ask something about your comment under the Promise<any> review, you said we should use something more concrete here instead of any, but that is Lambda manager, and we are very uncertain what will be returned here, hence doesn't `any` make sense here?

**Comment by bsekachev** [comment]
I believe we are certain enough. Otherwise how would it be possible to work with this API? 
As far as I remember `run()` returns a string, representing rq identificator to fetch statuses later. 
`call()` is more complex, but, more or less, you may find it out from this file `cvat-ui/src/components/annotation-page/standard-workspace/controls-side-bar/tools-control.tsx`. 

Perhaps the output is different for trackers, detectors and interactors. 
So, it should be like `Promise<TrackerResults | DetectorResults | InteractorResults>`, and define these three types additionally.

**Comment by tahamukhtar20** [comment]
Hi @bsekachev, I've added the required types and currently added the interfaces on top of the Lambda Manager file, please let me know if that is fine or if there is some other place more appropriate for the interfaces.
Additionally, I have updated the concerned code related to the lambda.call function, but I had to make some type assertions(e.g. number[] and number[][] in the tools-control.tsx file) to make it work with the existing code without making too many changes. Please let me know if this is the correct implementation for this.
Thank you.

File: cvat-core/src/lambda-manager.ts [hunk 3]
```diff
@@ -6,14 +6,36 @@
...
 export interface ModelProvider {
     name: string;
     icon: string;
     attributes: Record<string, string>;
 }
 
+export interface InteractorResults {
+    mask: number[][];
+    points: number[][] | number[];
+    bounds: [number, number, number, number]
```
Comment (bsekachev) [inline cvat-core/src/lambda-manager.ts:20]:
```suggestion
    points?: [number, number][];
    bounds?: [number, number, number, number]
```

File: cvat-core/src/lambda-manager.ts [hunk 4]
```diff
@@ -6,14 +6,36 @@
...
+    label: string;
+    outside?: boolean;
+    points?: number[];
+    mask?: number[];
+    elements: SerializedShape[];
+}
+
+export interface TrackerResults {
+    states: any[]
+    shapes: number[][]
```
Comment (bsekachev) [inline cvat-core/src/lambda-manager.ts:36]:
```suggestion
    states: any[];
    shapes: number[][];
```

File: cvat-core/src/lambda-manager.ts [hunk 5]
```diff
@@ -6,14 +6,36 @@
...
+export interface SerializedShape {
+    type: ShapeType;
+    rotation?: number;
+    attributes: { name: string; value: string }[];
+    label: string;
+    outside?: boolean;
+    points?: number[];
+    mask?: number[];
+    elements: SerializedShape[];
+}
```
Comment (bsekachev) [inline cvat-core/src/lambda-manager.ts:32]:
```suggestion
export interface DetectedShape {
    type: ShapeType;
    rotation?: number;
    attributes: { name: string; value: string }[];
    label: string;
    outside?: boolean;
    points?: number[];
    mask?: number[];
    elements: DetectedShape[];
}
```

File: cvat-core/src/lambda-manager.ts [hunk 6]
```diff
@@ -68,7 +90,7 @@ class LambdaManager {
         return result.id;
     }
 
-    async call(taskID, model, args) {
+    async call(taskID, model, args): Promise<TrackerResults | InteractorResults | SerializedShape[]> {
```
Comment (bsekachev) [inline cvat-core/src/lambda-manager.ts:93]:
```suggestion
    async call(taskID, model, args): Promise<TrackerResults | InteractorResults | DetectedShape[]> {
```

**Comment by bsekachev** [comment]
Please, avoid changes in `tools-control.tsx`, I will work with types inconsistency on my own, I see there are some issues.

**Comment by tahamukhtar20** [comment]
@bsekachev Updated

**Review by bsekachev — APPROVED** [review]

**Comment by bsekachev** [comment]
Just FYI: https://github.com/cvat-ai/cvat/pull/7785

**Comment by tahamukhtar20** [comment]
I found some more typescript warnings because of missing return types, maybe I can create a follow-up pull request for this if you allow it.

**Pull request merged**